### ISSUE TITLE
Admin is SUPER slow

### ISSUE THREAD

**Issue by jsciabica** [woothemes/woocommerce#9942]
Especially when searching orders.

	
Jonathan Sciabica
Dec 29, 00:23

I have just migrated from Magento to WooCommerce. So far I LOVE WooCommerce, but have found that pulling up a customer's order takes a really long time. I checked with my host and they said that the SQL requests are taking up to 17 seconds!

Here's what my host said:

I carefully investigated the server logs and found out that WooCommerce orders are viewed slowly in your website's backend because of slow MySQL queries toward your website's database - ooshine1_wp869.

Below I am providing you with the top 10 slowest recent MySQL queries:

Code:
1. Executed 1d 32m 1s ago for 17.505234 sec on Database --> ooshine1_wp869 
Date: 2015-12-21 17:26:06 Query_time: 17.505234 Rows_examined: 4607487: Rows_sent 0 Lock_time: 0.000166
SELECT DISTINCT p1.post_id FROM 8ez_postmeta p1 INNER JOIN 8ez_postmeta p2 ON p1.post_id = p2.post_id WHERE ( p1.meta_key = '_billing_first_name' AND p2.meta_key = '_billing_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%40906%' ) OR ( p1.meta_key = '_shipping_first_name' AND p2.meta_key = '_shipping_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%40906%' ) OR ( p1.meta_key IN ('_order_key','_billing_company','_billing_address_1','_billing_address_2','_billing_city','_billing_postcode','_billing_country','_billing_state','_billing_email','_billing_phone','_shipping_address_1','_shipping_address_2','_shipping_city','_shipping_postcode','_shipping_country','_shipping_state') AND p1.meta_value LIKE '%40906%' ); 
------------------------------------------------------------------------------------------------------- 
2. Executed 1h 14m 36s ago for 17.088139 sec on Database --> ooshine1_wp869 
Date: 2015-12-22 16:43:31 Query_time: 17.088139 Rows_examined: 4613509: Rows_sent 1 Lock_time: 0.000232 
SELECT DISTINCT p1.post_id FROM 8ez_postmeta p1 INNER JOIN 8ez_postmeta p2 ON p1.post_id = p2.post_id WHERE ( p1.meta_key = '_billing_first_name' AND p2.meta_key = '_billing_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%40741%' ) OR ( p1.meta_key = '_shipping_first_name' AND p2.meta_key = '_shipping_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%40741%' ) OR ( p1.meta_key IN ('_order_key','_billing_company','_billing_address_1','_billing_address_2','_billing_city','_billing_postcode','_billing_country','_billing_state','_billing_email','_billing_phone','_shipping_address_1','_shipping_address_2','_shipping_city','_shipping_postcode','_shipping_country','_shipping_state') AND p1.meta_value LIKE '%40741%' ); 
------------------------------------------------------------------------------------------------------- 
3. Executed 1d 32m 52s ago for 16.803624 sec on Database --> ooshine1_wp869 
Date: 2015-12-21 17:25:15 Query_time: 16.803624 Rows_examined: 4607340: Rows_sent 0 Lock_time: 0.000169 
SELECT DISTINCT p1.post_id FROM 8ez_postmeta p1 INNER JOIN 8ez_postmeta p2 ON p1.post_id = p2.post_id WHERE ( p1.meta_key = '_billing_first_name' AND p2.meta_key = '_billing_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%Betsy Shafe40906%' ) OR ( p1.meta_key = '_shipping_first_name' AND p2.meta_key = '_shipping_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%Betsy Shafe40906%' ) OR ( p1.meta_key IN ('_order_key','_billing_company','_billing_address_1','_billing_address_2','_billing_city','_billing_postcode','_billing_country','_billing_state','_billing_email','_billing_phone','_shipping_address_1','_shipping_address_2','_shipping_city','_shipping_postcode','_shipping_country','_shipping_state') AND p1.meta_value LIKE '%Betsy Shafe40906%' ); 
------------------------------------------------------------------------------------------------------- 
4. Executed 1d 32m 1s ago for 16.065079 sec on Database --> ooshine1_wp869 
Date: 2015-12-21 17:26:06 Query_time: 16.065079 Rows_examined: 4607487: Rows_sent 0 Lock_time: 0.000154 
SELECT DISTINCT p1.post_id FROM 8ez_postmeta p1 INNER JOIN 8ez_postmeta p2 ON p1.post_id = p2.post_id WHERE ( p1.meta_key = '_billing_first_name' AND p2.meta_key = '_billing_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%40906%' ) OR ( p1.meta_key = '_shipping_first_name' AND p2.meta_key = '_shipping_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%40906%' ) OR ( p1.meta_key IN ('_order_key','_billing_company','_billing_address_1','_billing_address_2','_billing_city','_billing_postcode','_billing_country','_billing_state','_billing_email','_billing_phone','_shipping_address_1','_shipping_address_2','_shipping_city','_shipping_postcode','_shipping_country','_shipping_state') AND p1.meta_value LIKE '%40906%' ); 
------------------------------------------------------------------------------------------------------- 
5. Executed 1h 53m 16s ago for 15.440493 sec on Database --> ooshine1_wp869 
Date: 2015-12-22 16:04:51 Query_time: 15.440493 Rows_examined: 4612822: Rows_sent 6 Lock_time: 0.000182 
SELECT DISTINCT p1.post_id FROM 8ez_postmeta p1 INNER JOIN 8ez_postmeta p2 ON p1.post_id = p2.post_id WHERE ( p1.meta_key = '_billing_first_name' AND p2.meta_key = '_billing_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%Richard Guerra%' ) OR ( p1.meta_key = '_shipping_first_name' AND p2.meta_key = '_shipping_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%Richard Guerra%' ) OR ( p1.meta_key IN ('_order_key','_billing_company','_billing_address_1','_billing_address_2','_billing_city','_billing_postcode','_billing_country','_billing_state','_billing_email','_billing_phone','_shipping_address_1','_shipping_address_2','_shipping_city','_shipping_postcode','_shipping_country','_shipping_state') AND p1.meta_value LIKE '%Richard Guerra%' ); 
------------------------------------------------------------------------------------------------------- 
6. Executed 2h 8m 17s ago for 15.28621 sec on Database --> ooshine1_wp869 
Date: 2015-12-22 15:49:50 Query_time: 15.286210 Rows_examined: 4613063: Rows_sent 0 Lock_time: 0.000227 
SELECT DISTINCT p1.post_id FROM 8ez_postmeta p1 INNER JOIN 8ez_postmeta p2 ON p1.post_id = p2.post_id WHERE ( p1.meta_key = '_billing_first_name' AND p2.meta_key = '_billing_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%40529%' ) OR ( p1.meta_key = '_shipping_first_name' AND p2.meta_key = '_shipping_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%40529%' ) OR ( p1.meta_key IN ('_order_key','_billing_company','_billing_address_1','_billing_address_2','_billing_city','_billing_postcode','_billing_country','_billing_state','_billing_email','_billing_phone','_shipping_address_1','_shipping_address_2','_shipping_city','_shipping_postcode','_shipping_country','_shipping_state') AND p1.meta_value LIKE '%40529%' ); 
------------------------------------------------------------------------------------------------------- 
7. Executed 1d 40m 6s ago for 15.240986 sec on Database --> ooshine1_wp869 
Date: 2015-12-21 17:18:01 Query_time: 15.240986 Rows_examined: 4607337: Rows_sent 0 Lock_time: 0.000138
SELECT DISTINCT p1.post_id FROM 8ez_postmeta p1 INNER JOIN 8ez_postmeta p2 ON p1.post_id = p2.post_id WHERE ( p1.meta_key = '_billing_first_name' AND p2.meta_key = '_billing_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%40906%' ) OR ( p1.meta_key = '_shipping_first_name' AND p2.meta_key = '_shipping_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%40906%' ) OR ( p1.meta_key IN ('_order_key','_billing_company','_billing_address_1','_billing_address_2','_billing_city','_billing_postcode','_billing_country','_billing_state','_billing_email','_billing_phone','_shipping_address_1','_shipping_address_2','_shipping_city','_shipping_postcode','_shipping_country','_shipping_state') AND p1.meta_value LIKE '%40906%' );
-------------------------------------------------------------------------------------------------------
8. Executed 1h 4m 29s ago for 15.238117 sec on Database --> ooshine1_wp869
Date: 2015-12-22 16:53:38 Query_time: 15.238117 Rows_examined: 4613667: Rows_sent 2 Lock_time: 0.000131
SELECT DISTINCT p1.post_id FROM 8ez_postmeta p1 INNER JOIN 8ez_postmeta p2 ON p1.post_id = p2.post_id WHERE ( p1.meta_key = '_billing_first_name' AND p2.meta_key = '_billing_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%40726%' ) OR ( p1.meta_key = '_shipping_first_name' AND p2.meta_key = '_shipping_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%40726%' ) OR ( p1.meta_key IN ('_order_key','_billing_company','_billing_address_1','_billing_address_2','_billing_city','_billing_postcode','_billing_country','_billing_state','_billing_email','_billing_phone','_shipping_address_1','_shipping_address_2','_shipping_city','_shipping_postcode','_shipping_country','_shipping_state') AND p1.meta_value LIKE '%40726%' );
-------------------------------------------------------------------------------------------------------
9. Executed 1d 32m 50s ago for 15.050109 sec on Database --> ooshine1_wp869
Date: 2015-12-21 17:25:17 Query_time: 15.050109 Rows_examined: 4607340: Rows_sent 0 Lock_time: 0.000249
SELECT DISTINCT p1.post_id FROM 8ez_postmeta p1 INNER JOIN 8ez_postmeta p2 ON p1.post_id = p2.post_id WHERE ( p1.meta_key = '_billing_first_name' AND p2.meta_key = '_billing_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%40906%' ) OR ( p1.meta_key = '_shipping_first_name' AND p2.meta_key = '_shipping_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%40906%' ) OR ( p1.meta_key IN ('_order_key','_billing_company','_billing_address_1','_billing_address_2','_billing_city','_billing_postcode','_billing_country','_billing_state','_billing_email','_billing_phone','_shipping_address_1','_shipping_address_2','_shipping_city','_shipping_postcode','_shipping_country','_shipping_state') AND p1.meta_value LIKE '%40906%' );
-------------------------------------------------------------------------------------------------------
10. Executed 2h 41m 44s ago for 14.694335 sec on Database --> ooshine1_wp869
Date: 2015-12-22 15:16:23 Query_time: 14.694335 Rows_examined: 4612485: Rows_sent 7 Lock_time: 0.000172
SELECT DISTINCT p1.post_id FROM 8ez_postmeta p1 INNER JOIN 8ez_postmeta p2 ON p1.post_id = p2.post_id WHERE ( p1.meta_key = '_billing_first_name' AND p2.meta_key = '_billing_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%40891%' ) OR ( p1.meta_key = '_shipping_first_name' AND p2.meta_key = '_shipping_last_name' AND CONCAT(p1.meta_value, ' ', p2.meta_value) LIKE '%40891%' ) OR ( p1.meta_key IN ('_order_key','_billing_company','_billing_address_1','_billing_address_2','_billing_city','_billing_postcode','_billing_country','_billing_state','_billing_email','_billing_phone','_shipping_address_1','_shipping_address_2','_shipping_city','_shipping_postcode','_shipping_country','_shipping_state') AND p1.meta_value LIKE '%40891%' );

As you can see, the queries above look like they are fetching information about orders from your website's database and take between 14 and 17 seconds to execute. I would like to clarify that normally MySQL queries should not take more than 1 second to execute.

In order to resolve this issue, you should optimize your database for better performance. Looking at the queries above I can see that most likely the issue is caused by the following:

Rows_examined: 4613667

Basically this means is that every time your WooCommerce tries to get information about an order(s) from the database it must examine over 4.5 million rows of data, which is very, very inefficient.

**Comment by mikejolley** [comment]
At it's heart, this is the main issue here: https://github.com/woothemes/woocommerce/issues/9735

However, since 40891 is numeric I think we can just search on order IDs to at least make those searches faster. Text based searches will still require the joins.

**Comment by jsciabica** [comment]
Thanks for the prompt feedback!

How about a dropdown menu next to the search field for "Order Number",
"Email", "Name", "All Fields"?  Or better yet, Magento has an input above
each field (see attached).  Simply type in the header column and press
enter.  Shows just the matching orders in about 1 second.

Is there anything I can do in my database to speed this up?


Best Regards,
Jonathan Sciabica

Sciabica's California Olive Oil
2150 Yosemite Blvd
Modesto, CA 95354
SunshineInABottle.com

**Comment by mikejolley** [comment]
You can use the 'optimize' tool in PHPMyAdmin as one example.  A dropdown is an option, but it would require something 100% custom rather than leveraging what WP already does.

**Comment by mikejolley** [comment]
Just committed an improvement to numeric (order ID) search.
